src/sentry/static/sentry/app/views/eventsV2/utils.tsx

```diff
@@ -376,3 +379,32 @@ export function decodeScalar(
       : undefined;
   return isString(unwrapped) ? unwrapped : undefined;
 }
+
+export function downloadAsCsv(tableData, columnOrder, filename) {
+  const {data} = tableData;
+  const headings = columnOrder.map(column => column.name);
+
+  const csvContent = Papa.unparse({
+    fields: headings,
+    data: data.map(row => {
+      return headings.map(col => {
+        // alias for project doesn't match the table data name
+        if (col === 'project') {
+          col = 'project.name';
+        } else {
+          col = getAggregateAlias(col);
+        }
+        return disableMacros(row[col]);
+      });
+    }),
+  });
+
+  const encodedDataUrl = encodeURI(`data:text/csv;charset=utf8,${csvContent}`);
+
+  // Create a download link then click it, this is so we can get a filename
+  const link = document.createElement('a');
+  const now = new Date();
+  link.setAttribute('href', encodedDataUrl);
+  link.setAttribute('download', `${filename} ${getUtcDateString(now)}.csv`);
+  link.click();
```

Comment:
@wmak does this work in FireFox and Safari?

Some variations of this appends the link to the DOM before clicking, and removing it afterwards:

- https://github.com/mholt/PapaParse/issues/175#issuecomment-201308792
- https://stackoverflow.com/questions/32225904/programmatical-click-on-a-tag-not-working-in-firefox

Reply:
:+1: Works in both, but I'll also add the removing.